xdsl/irdl/irdl.py

```diff
@@ -2465,15 +2465,16 @@ def irdl_attr_definition(cls: TypeAttributeInvT) -> TypeAttributeInvT:
     if issubclass(cls, ParametrizedAttribute):
         return irdl_param_attr_definition(cls)
     if issubclass(cls, Data):
-        return runtime_final(
-            dataclass(frozen=True)(  # pyright: ignore[reportGeneralTypeIssues]
-                type(
-                    cls.__name__,
-                    (cls,),  # pyright: ignore[reportUnknownArgumentType]
-                    dict(cls.__dict__),
-                )
+        data_class = cast(type[Data[Any]], cls)
+        data_metaclass = cast(ABCMeta, data_class.__class__)
+        irdled = runtime_final(
+            data_metaclass(
+                cls.__name__,
+                (cls, *cls.__bases__),
+                dict(cls.__dict__),
```

Comment:
Can you add a comment somewhere that explains what this does, for instance that it will make the class frozen because of the dataclass?

Reply:
Thank you for discussing this;
Explaining it, I realized that this was now only rebuilding the class itself as-is... There is no need for a dynamic class at all :yum: 
Added a comment!